src/lib.rs

```diff
@@ -89,16 +82,15 @@ impl EncodingDetector {
 
     /// Pass a chunk of raw bytes to the detector. This is a no-op if a
     /// charset has been detected.
-    fn handle_data(&mut self, data: &[u8]) -> EncodingDetectorResult<()> {
+    fn handle_data(&mut self, data: &[u8]) -> Result<()> {
         let result = unsafe {
             ffi::uchardet_handle_data(self.ptr, data.as_ptr() as *const i8,
                                       data.len() as size_t)
         };
         match result {
             0 => Ok(()),
             _ => {
-                let msg =  "Error handling data".to_string();
-                Err(EncodingDetectorError{message: msg})
+                Err(ErrorKind::DataHandlingError.into())
```

Comment:
OK, in this case, we should map `nsresult` to `Ok(())`, `Err(ErrorKind::OutOfMemory)` and `Err(ErrorKind::Other)`. We might want to do this using an `impl From<nsresult> for ErrorKind` in the `errors` module if `nsresult` is an actual `struct` or `enum` in Rust. Or if `nsresult` is just a `type` alias, then we could define a private `ErrorKind::from_nsresult` function.